src/dependency-manager/src/service-config/v1.ts

```diff
@@ -1,142 +1,366 @@
 import { plainToClass } from 'class-transformer';
 import { Transform, Type } from 'class-transformer/decorators';
-import { ParameterValue } from '../manager';
+import { IsBoolean, IsEmpty, IsIn, IsInstance, IsNotEmpty, IsNumber, IsOptional, IsString, Matches, ValidateIf, ValidatorOptions } from 'class-validator';
+import { BaseSpec } from '../utils/base-spec';
+import { Dictionary } from '../utils/dictionary';
 import { Dict } from '../utils/transform';
+import { validateDictionary, validateNested } from '../utils/validation';
+import { ParameterDefinitionSpecV1 } from '../v1-spec/parameters';
 import { ServiceApiSpec, ServiceConfig, ServiceDatastore, ServiceDebugOptions, ServiceEventNotifications, ServiceEventSubscriptions, ServiceInterfaceSpec, ServiceParameter, VolumeSpec } from './base';
 
-function transformParameters(input: any) {
-  const output: any = {};
+const transformParameters = (input?: Dictionary<any>): Dictionary<ParameterDefinitionSpecV1> | undefined => {
+  if (!input) {
```

Comment:
Maybe change these checks for undefined explictly? Don't want to throw away empty string/0

Reply:
Maybe even warrants a test case

Reply:
Do we do something different for `parameters: ""` than for not having a parameters field at all?